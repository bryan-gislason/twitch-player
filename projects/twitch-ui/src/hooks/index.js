export { default as useDebug } from "./useDebug";
export { default as useDimensions } from "./useDimensions";
export * from "./useGridList";
export * from "./useHorizontalList";
export { default as useKeyboard } from "./useKeyboard";
export { default as useLanguage } from "./useLanguage";
export { default as useOverrides } from "./useOverrides";
export { default as useRouter } from "./useRouter";
