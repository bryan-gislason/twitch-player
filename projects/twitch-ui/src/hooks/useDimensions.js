import { throttle } from "@twitch-player/utils";
import { useCallback, useContext } from "react";
import { DimensionsContext } from "../context";

const useDimensions = key => {
  const [dimensions, setDimensions] = useContext(DimensionsContext);

  const updateDimensionsCallback = useCallback(
    throttle(({ nativeEvent }) => {
      setDimensions(prevState => ({ ...prevState, [key]: nativeEvent.layout }));
    }),
    [key]
  );

  return [dimensions[key], updateDimensionsCallback];
};

export default useDimensions;
