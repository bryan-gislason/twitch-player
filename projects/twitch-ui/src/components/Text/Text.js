import { useTheme } from "@twitch-player/themes/dist/hooks";
import PropTypes from "prop-types";
import React, { memo } from "react";
import { Text } from "react-native";

const CustomText = ({ style = [], ...props }) => {
  const [{ typography }] = useTheme();

  const rootStyle = [
    {
      fontFamily: typography.family,
      lineHeight: `${typography.lineHeight}`,
    },
  ].concat(style);

  return <Text {...props} style={rootStyle} />;
};

CustomText.propTypes = {
  style: PropTypes.any,
};

export default memo(CustomText);
