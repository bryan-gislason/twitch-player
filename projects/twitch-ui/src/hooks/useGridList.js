import { useTheme } from "@twitch-player/themes/dist/hooks";
import {
  getItemDimensions,
  getGridListItemLayout,
  getGridListItemStyle,
} from "../utils";
import { useDimensions } from ".";

export const useCategoryGrid = itemCount => {
  const [{ layout, typography }] = useTheme();
  const [{ width, height }] = useDimensions("main");

  const options = {
    itemCount,
    containerWidth: width,
    containerHeight: height,
    maxImageWidth: layout.maxCategoryWidth,
    maxImageHeight: layout.maxCategoryHeight,
    detailHeight:
      Math.floor(typography.sizeSecondary * typography.lineHeight) +
      Math.floor(typography.sizeTertiary * typography.lineHeight) +
      layout.gapSmall * 2,
    spaceSides: layout.gapExtraLarge,
    spaceBetween: layout.gapSmall,
    spaceBottom: layout.gapMedium,
  };

  const itemDimensions = getItemDimensions(options);
  const extendedOptions = { ...options, ...itemDimensions };

  const getItemLayout = itemCount && getGridListItemLayout(extendedOptions);
  const renderItemStyle = itemCount && getGridListItemStyle(extendedOptions);

  return {
    ...extendedOptions,
    getItemLayout,
    renderItemStyle,
  };
};

export const useChannelGrid = itemCount => {
  const [{ layout, typography }] = useTheme();
  const [{ width, height }] = useDimensions("main");

  const options = {
    itemCount,
    containerWidth: width,
    containerHeight: height,
    maxImageWidth: layout.maxChannelWidth,
    maxImageHeight: layout.maxChannelHeight,
    detailHeight:
      Math.floor(typography.sizeSecondary * typography.lineHeight) +
      Math.floor(typography.sizeTertiary * typography.lineHeight) +
      Math.floor(typography.sizeTertiary * typography.lineHeight) +
      layout.gapSmall * 2,
    spaceSides: layout.gapExtraLarge,
    spaceBetween: layout.gapSmall,
    spaceBottom: layout.gapMedium,
  };

  const itemDimensions = getItemDimensions(options);
  const extendedOptions = { ...options, ...itemDimensions };

  const getItemLayout = itemCount && getGridListItemLayout(extendedOptions);
  const renderItemStyle = itemCount && getGridListItemStyle(extendedOptions);

  return {
    ...extendedOptions,
    getItemLayout,
    renderItemStyle,
  };
};
