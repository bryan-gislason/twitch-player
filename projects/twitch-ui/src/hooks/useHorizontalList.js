import { useTheme } from "@twitch-player/themes/dist/hooks";
import {
  getItemDimensions,
  getHorizontalListItemLayout,
  getHorizontalListItemStyle,
} from "../utils";
import { useDimensions } from ".";

export const useCategoryList = itemCount => {
  const [{ layout }] = useTheme();
  const [{ width }] = useDimensions("main");

  const options = {
    itemCount,
    containerWidth: width,
    maxImageWidth: layout.maxCategoryWidth,
    maxImageHeight: layout.maxCategoryHeight,
    spaceSides: layout.gapExtraLarge,
    spaceBetween: layout.gapSmall,
    spaceBottom: layout.gapMedium,
  };

  const itemDimensions = getItemDimensions(options);
  const extendedOptions = { ...options, ...itemDimensions };

  const getItemLayout =
    itemCount && getHorizontalListItemLayout(extendedOptions);
  const renderItemStyle =
    itemCount && getHorizontalListItemStyle(extendedOptions);

  return {
    ...extendedOptions,
    getItemLayout,
    renderItemStyle,
  };
};

export const useChannelList = itemCount => {
  const [{ layout }] = useTheme();
  const [{ width }] = useDimensions("main");

  const options = {
    itemCount,
    containerWidth: width,
    maxImageWidth: layout.maxChannelWidth,
    maxImageHeight: layout.maxChannelHeight,
    spaceSides: layout.gapExtraLarge,
    spaceBetween: layout.gapSmall,
    spaceBottom: layout.gapMedium,
  };

  const itemDimensions = getItemDimensions(options);
  const extendedOptions = { ...options, ...itemDimensions };

  const getItemLayout =
    itemCount && getHorizontalListItemLayout(extendedOptions);
  const renderItemStyle =
    itemCount && getHorizontalListItemStyle(extendedOptions);

  return {
    ...extendedOptions,
    getItemLayout,
    renderItemStyle,
  };
};
