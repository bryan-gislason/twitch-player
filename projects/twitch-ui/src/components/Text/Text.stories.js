import { storiesOf } from "@storybook/react";
import React from "react";
import Text from ".";

storiesOf("Static Components|Text", module).add("default", () => (
  <Text>OHAI</Text>
));
