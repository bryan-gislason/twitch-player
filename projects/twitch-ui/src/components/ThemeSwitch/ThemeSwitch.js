import { useTranslation } from "@twitch-player/i18n";
import { useTheme } from "@twitch-player/themes/dist/hooks";
import {
  dark as darkTheme,
  light as lightTheme,
} from "@twitch-player/themes/dist/themes";
import PropTypes from "prop-types";
import React, { memo } from "react";
import Switch from "../Switch";
import Text from "../Text";

const ThemeSwitch = ({ testID = "theme-switch" }) => {
  const { t } = useTranslation();
  const [theme, setTheme] = useTheme();

  const isDarkTheme = theme === darkTheme;
  const toggleDarkTheme = value => setTheme(value ? darkTheme : lightTheme);

  const textStyle = {
    color: theme.colors.bodyText,
    fontSize: theme.typography.sizeSecondary,
  };

  return (
    <Text accessibilityRole="label" testID={testID}>
      <Text style={textStyle}>{t("pages.settings.themeHeading")}</Text>
      <Switch value={isDarkTheme} onValueChange={toggleDarkTheme} />
    </Text>
  );
};

ThemeSwitch.propTypes = {
  testID: PropTypes.string,
};

export default memo(ThemeSwitch);
