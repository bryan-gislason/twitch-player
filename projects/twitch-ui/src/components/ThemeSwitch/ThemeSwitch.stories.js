import { storiesOf } from "@storybook/react";
import React from "react";
import ThemeSwitch from ".";

storiesOf("Components|ThemeSwitch", module).add("default", () => (
  <ThemeSwitch />
));
