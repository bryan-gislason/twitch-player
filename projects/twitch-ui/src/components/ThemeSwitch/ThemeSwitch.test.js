import { useTheme } from "@twitch-player/themes/dist/hooks";
import React from "react";
import { fireEvent, render } from "../../testing";
import Text from "../Text";
import ThemeSwitch from "./ThemeSwitch";

const SomeComponent = () => {
  const [{ colors }] = useTheme();

  return <Text testID="some-component">{colors.bodyText}</Text>;
};

describe("ThemeSwitch", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <>
        <ThemeSwitch />
        <SomeComponent />
      </>
    );
  });

  it("renders a theme switch", () => {
    expect(instance.getByTestId("theme-switch")).toMatchSnapshot();
  });

  it("renders with the default theme", () => {
    expect(instance.getByTestId("some-component")).toMatchSnapshot();
  });

  describe("when clicked", () => {
    let checkbox;

    beforeEach(() => {
      checkbox = instance.getByLabelText("Theme", { selector: "input" });
      fireEvent.click(checkbox);
    });

    it("re-renders with the new theme", () => {
      expect(instance.getByTestId("some-component")).toMatchSnapshot();
    });

    describe("when clicked again", () => {
      beforeEach(() => {
        fireEvent.click(checkbox);
      });

      it("re-renders with the default theme", () => {
        expect(instance.getByTestId("some-component")).toMatchSnapshot();
      });
    });
  });
});
