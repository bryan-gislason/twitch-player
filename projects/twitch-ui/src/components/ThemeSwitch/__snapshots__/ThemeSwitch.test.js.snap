// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch renders a theme switch 1`] = `
<label
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="theme-switch"
  dir="auto"
  style="font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; line-height: 1.5;"
>
  <span
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-irrty rn-wordWrap-qvutc0"
    dir="auto"
    style="color: rgb(25, 23, 28); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 16px; line-height: 1.5;"
  >
    Theme
  </span>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-xoduu5 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-userSelect-lrvibr rn-zIndex-1lgpqti"
    style="height: 20px; width: 40px;"
  >
    <div
      class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-bottom-1p0dtai rn-boxSizing-deolkf rn-display-xoduu5 rn-flexDirection-eqz5dr rn-height-1dernwh rn-left-1d2f490 rn-marginTop-1t01tom rn-marginRight-lchren rn-marginBottom-1qahzrx rn-marginLeft-1jj8364 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-transitionDuration-13tjlyg rn-width-13qz1uu rn-zIndex-1lgpqti"
      style="background-color: rgb(218, 216, 222); border-top-left-radius: 10px; border-top-right-radius: 10px; border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
    />
    <div
      class="rn-alignItems-1oszu61 rn-alignSelf-k200y rn-borderTopLeftRadius-jt3ufn rn-borderTopRightRadius-1e868j9 rn-borderBottomRightRadius-ujv9e3 rn-borderBottomLeftRadius-1hakmuk rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxShadow-1ewcgjf rn-boxSizing-deolkf rn-display-xoduu5 rn-flexDirection-eqz5dr rn-left-1fe0xdi rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-transform-emqnss rn-transitionDuration-13tjlyg rn-zIndex-1lgpqti"
      style="background-color: rgb(250, 249, 250); height: 20px; width: 20px;"
    />
    <input
      class="rn-bottom-1p0dtai rn-cursor-1ei5mc7 rn-height-1pi2tsx rn-left-1d2f490 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-opacity-1272l3b rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu"
      data-focusable="true"
      type="checkbox"
    />
  </div>
</label>
`;

exports[`ThemeSwitch renders with the default theme 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="some-component"
  dir="auto"
  style="font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; line-height: 1.5;"
>
  #19171c
</div>
`;

exports[`ThemeSwitch when clicked re-renders with the new theme 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="some-component"
  dir="auto"
  style="font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; line-height: 1.5;"
>
  #faf9fa
</div>
`;

exports[`ThemeSwitch when clicked when clicked again re-renders with the default theme 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="some-component"
  dir="auto"
  style="font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; line-height: 1.5;"
>
  #19171c
</div>
`;
